components/movementsensor/gpsnmea/pmtkI2C.go

```diff
@@ -256,7 +262,22 @@ func (g *PmtkI2CNMEAMovementSensor) AngularVelocity(
 func (g *PmtkI2CNMEAMovementSensor) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
 	g.mu.RLock()
 	defer g.mu.RUnlock()
-	return 0, g.err.Get()
+	currentHeading := g.data.CompassHeading
+	lastHeading := g.lastcompassheading.GetLastCompassHeading()
+
+	if !math.IsNaN(currentHeading) && !math.IsNaN(lastHeading) && currentHeading != lastHeading {
+		g.lastcompassheading.SetLastCompassHeading(g.data.CompassHeading)
+		currentHeading = lastHeading
```

Comment:
hmm, I am already updating lastHeading by calling SetLastCompassHeading. I should just remove setting currHeading as lastHeading since it will be valid.